src/extendStyles.jsx

```diff
@@ -0,0 +1,45 @@
+/* eslint react/forbid-foreign-prop-types: off */
+
+import deepmerge from 'deepmerge';
+
+// Recursively iterate through the style object, validating the same path exists in the
+// extendableStyles object.
+function validateStyle(style, extendableStyles, path = '') {
+  if (process.env.NODE_ENV !== 'production') {
+    // Stop recursively validating when we hit a style's value
+    if (!style || Array.isArray(style) || typeof style !== 'object') {
+      return;
+    }
+
+    const styleKeys = Object.keys(style);
+    if (styleKeys.length > 0) {
+      styleKeys.forEach((styleKey) => {
+        const currentPath = `${path}.${styleKey}`;
+        const isValid = extendableStyles[styleKey];
```

Comment:
It’s\d be great if instead of a boolean, this was a predicate function - then the *values* can be constrained as well, if desired, and if not, `() => true` can be provided?

Reply:
done!
https://github.com/airbnb/react-with-styles/pull/210/commits/0dc71f44fbb958246f106f50bbd13504869c2901

Reply:
and https://github.com/airbnb/react-with-styles/pull/210/commits/8a185d02a48e5d84aff541189598f5daca68db5c